components/prism-qsharp.js

```diff
@@ -0,0 +1,145 @@
+	} 
+	// keywords
+	function keywordsToPattern(words) {
+		return '\\b(?:' + words.trim().replace(/ /g, '|') + ')\\b';
+	}
+	var keywords = RegExp(keywordsToPattern(keywordKinds.type + ' ' + keywordKinds.other));
+
+	// types
+	var generic = nested(/<(?:[^<>;=+\-*/%&|^]|<<self>>)*>/.source, 2);
+	var name = /\b[A-Za-z_]\w*\b/.source;
+	var genericName = replace(/<<0>>(?:\s*<<1>>)?/.source, [name, generic]);
+	var identifier = replace(/(?!<<0>>)<<1>>(?:\s*\.\s*<<1>>)*/.source, [keywordsToPattern(keywordKinds.other), genericName]);
+
+	var typeInside = {
+		'keyword': keywords,
+		'punctuation': /[<>()?,.:[\]]/
+	};
+
+	// strings
+	var regularString = /"(?:\\.|[^\\"\r\n])*"/.source;
+
+	Prism.languages.qsharp = Prism.languages.extend('clike', {
+		'comment': /\/\/.*/,
+		'string': [
+			{
+				pattern: re(/(^|[^$\\])<<0>>/.source, [regularString]),
+				lookbehind: true,
+				greedy: true
+			}
+		],
+		'class-name': [
+			{
+				// open Microsoft.Quantum.Canon;
+				pattern: re(/(\b(?:as|open)\s+)<<0>>(?=\s*(?:;|as\b))/.source, [identifier]),
+				lookbehind: true,
+				inside: typeInside
+			},
+			{
+				// namespace Quantum.App1;
+				pattern: re(/(\bnamespace\s+)<<0>>(?=\s*{)/.source, [identifier]),
+				lookbehind: true,
+				inside: typeInside
+			},
+		],
+		'keyword': keywords,
+		'number': /(?:\b0(?:x[\da-f]+|b[01]+|o[0-7]+)|(?:\B\.\d+|\b\d+(?:\.\d*)?)(?:e[-+]?\d+)?)l?\b/i,
+		'operator': /\band=|\bor=|\band\b|\bor\b|\bnot\b|<[-=]|[-=]>|>>>=?|<<<=?|\^\^\^=?|\|\|\|=?|&&&=?|w\/=?|~~~|[*\/+\-^=!%]=?/,
+		'punctuation': /::|[{}[\];(),.:]/
+	});
+
+	Prism.languages.insertBefore('qsharp', 'number', {
+		'range': {
+			pattern: /\.\./,
+			alias: 'operator'
+		}
+	});
+
+	// string interpolation
+	var formatString = /:[^}\r\n]+/.source;
+
+	// single line
+	var sInterpolationRound = nested(replace(/[^"()]|<<0>>|\(<<self>>*\)/.source, [regularString]), 2)
+	var sInterpolation = replace(/\{(?!\{)(?:(?![}:])<<0>>)*<<1>>?\}/.source, [sInterpolationRound, formatString]);
+
+	Prism.languages.insertBefore('qsharp', 'string', {
+		'interpolation-string': {
+			pattern: re(/(^|[^@\\])\$"(?:\\.|\{\{|<<0>>|[^\\"{])*"/.source, [sInterpolation]),
+			lookbehind: true,
+			greedy: true,
+			inside: {
+				'interpolation': {
+					pattern: re(/((?:^|[^{])(?:\{\{)*)<<0>>/.source, [sInterpolation]),
+					lookbehind: true,
+					inside: {
+						'format-string': {
+							pattern: re(/(^\{(?:(?![}:])<<0>>)*)<<1>>(?=\}$)/.source, [sInterpolationRound, formatString]),
+							lookbehind: true,
+							inside: {
+								'punctuation': /^:/
+							}
+						},
+						'punctuation': /^\{|\}$/,
+						'expression': {
+							pattern: /[\s\S]+/,
+							alias: 'language-qsharp',
+							inside: Prism.languages.qsharp
+						}
+					}
+				},
+				'string': /[\s\S]+/
+			}
+		}
+	});
```

Comment:
Q#'s string interpolation grammar is a lot simpler than C#'s one, so we can simplify this.

```suggestion
	// single line
	var interpolationExpr = nested(replace(/\{(?:[^"{}]|<<0>>|<<self>>)*\}/.source, [regularString]), 2);

	Prism.languages.insertBefore('qsharp', 'string', {
		'interpolation-string': {
			pattern: re(/\$"(?:\\.|<<0>>|[^\\"{])*"/.source, [interpolationExpr]),
			greedy: true,
			inside: {
				'interpolation': {
					pattern: re(/((?:^|[^\\])(?:\\\\)*)<<0>>/.source, [interpolationExpr]),
					lookbehind: true,
					inside: {
						'punctuation': /^\{|\}$/,
						'expression': {
							pattern: /[\s\S]+/,
							alias: 'language-qsharp',
							inside: Prism.languages.qsharp
						}
					}
				},
				'string': /[\s\S]+/
			}
		}
	});
```